trade-finance-manager-ui/scripts/enable-select-all-table-checkbox.js

```diff
@@ -0,0 +1,55 @@
+const selectAllTableCellCheckboxSelector = 'table.govuk-table input[type="checkbox"]#select-all-checkbox';
+const selectAllTableCellCheckbox = document.querySelector(selectAllTableCellCheckboxSelector);
+
+const allOtherTableCellCheckboxesSelector = 'table.govuk-table input[type="checkbox"]:not(#select-all-checkbox)';
+const allOtherTableCellCheckboxes = document.querySelectorAll(allOtherTableCellCheckboxesSelector);
```

Comment:
`querySelectorAll` returns a `NodeList`. In chrome, it is possible to extract an array from this type using `document.querySelectorAll(...).entries().toArray()`. However, this `.toArray` method does not appear in my intellisense in VSCode, which leads me to believe this behaviour may not be supported in all browsers. To be on the safe side, I went with the `Array` prototype methods below which I believe have very good browser support, but I'm not sure if this is something we necessarily want to do

Reply:
If this works I think it is fine, it is pretty neat and sensible to think about the browser support. We could also just use a `forEach` over the `NodeList` with some additional logic instead of the array `all` and `some` methods but I don't know that it matters too much?

Reply:
I tend to prefer using `some` and `every` over a `forEach` because (as far as I know) you can't get it to stop early without wrapping it in a try-catch and making it throw, whereas those two will stop early if they don't meet the criteria